src/transformers/generation/utils.py

```diff
@@ -4616,6 +4603,7 @@ def assisted_decoding(
                 else:
                     last_assistant_token_is_eos = False
 
+            model_kwargs["assistant_past_key_values"] = assistant_kwargs["assistant_past_key_values"]
```

Comment:
We can remove this line if we change 

```
            model_kwargs["assistant_past_key_values"] = _crop_past_key_values(
                assistant_model, model_kwargs["assistant_past_key_values"], new_cache_size - 1
            )  # the assistant does not have the token after the last match, hence the -1
```

a few lines below into

```
            assistant_kwargs["assistant_past_key_values"] = _crop_past_key_values(
                assistant_model, assistant_kwargs["assistant_past_key_values"], new_cache_size - 1
            )  # the assistant does not have the token after the last match, hence the -1
```

That way, we would have no assistant kwargs in `model_kwargs`, which would be great!

Reply:
It is not possible. `model_kwargs` should always have `assistant_past_key_values` because we copied `assistant_kwargs` from `model_kwargs` in each round.

Reply:
Oh, I see your point. There is no need to copy `model_kwargs` in every round.